pkg/block/azure/adapter.go

```diff
@@ -193,8 +198,11 @@ func (a *Adapter) Download(ctx context.Context, obj block.ObjectPointer, offset,
 
 	keyOptions := azblob.ClientProvidedKeyOptions{}
 	downloadResponse, err := blobURL.Download(ctx, offset, count, azblob.BlobAccessConditions{}, false, keyOptions)
-
+	if isErrNotFound(err) {
+		return nil, adapter.ErrNotFound
+	}
 	if err != nil {
+		a.log(ctx).WithError(err).Errorf("failed to get azure blob from container %s key %s", container, blobURL)
```

Comment:
Usually errors should be logged at the topmost level -- more context is available there (e.g. because intermediate levels have wrapped the error).